packages/commerce-sdk-react/src/hooks/useAuthorizationHeader.ts

```diff
@@ -25,6 +31,27 @@ export const useAuthorizationHeader = <Options extends ApiOptions, Data>(
                 Authorization: `Bearer ${access_token}`,
                 ...options.headers
             }
+        }).catch(async (error) => {
+            if (error?.response?.status == 401) {
+                const response = await error?.response?.json()
+                if (response?.detail === 'Customer credentials changed after token was issued.') {
+                    logger.info('Login was invalidated. Clearing login state.')
+                    const {access_token} = await auth.logout()
+
+                    // Retry again after resetting auth state
+                    return await method({
+                        ...options,
+                        headers: {
+                            Authorization: `Bearer ${access_token}`,
+                            ...options.headers
+                        }
+                    })
+                } else {
+                    throw error
+                }
+            } else {
+                throw error
+            }
```

Comment:
consider invert the `if`s, so the code has less nested statements

```js
if (error?.response?.status !== 401) {
    throw error;
}

const response = await error?.response?.json();
if (response?.detail !== 'Customer credentials changed after token was issued.') {
    throw error;
}

logger.info('Login was invalidated. Clearing login state.');
const {access_token} = await auth.logout();

// Retry again after resetting auth state
return await method({
    ...options,
    headers: {
        Authorization: `Bearer ${access_token}`,
        ...options.headers
    }
});
```

Reply:
This suggestion has been applied